fe/src/main/java/org/apache/doris/system/Backend.java

```diff
@@ -346,6 +347,23 @@ public double getMaxDiskUsedPct() {
         return maxPct;
     }
 
+    public boolean checkDiskExceedLimitByStorageMedium(TStorageMedium storageMedium) {
+        // no available disk with storage medium
+        if (getDiskNumByStorageMedium(storageMedium) <= 0) {
+            return true;
+        }
+        ImmutableMap<String, DiskInfo> disks = disksRef.get();
+        boolean exceedLimit = false;
+        for (DiskInfo diskInfo : disks.values()) {
```

Comment:
Is this logic correct? If all disks are offline or is not with specified storage medium, then this method will return false?

Reply:
yes, you are right, I fixed it, and should we change the logic in be, just remove the judgement `_available_storage_medium_type_count == 1`?

```
for (auto& it : _store_map) {
            if (it.second->is_used()) {
                if (_available_storage_medium_type_count == 1
                    || it.second->storage_medium() == storage_medium) {
                    stores.push_back(it.second);
                }
            }
        }
```